### PR TITLE
Compile with mingw on Windows.

### PR DESCRIPTION
**Pull request by djhaskin987**
This PR allows `rep` to be compiled with mingw on Windows. I have printed help page, and successfully connected to a repl with the compiled result. 

To compile, simply run:

```
mingw32-make.exe
```

from the root directory of the repository.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Code
```
Comment (djhaskin987) [inline .gitattributes:1]: .gitattributes needed so that git can handle the different line endings between windows and linux.

File: .gitignore [hunk 2]
```diff
@@ -14,6 +14,7 @@ pom.xml.asc
 /result
 /.cpcache
 /rep
+/rep.exe
```
Comment (djhaskin987) [inline .gitignore:17]: Add `rep` equivalent entry for windows.

File: Makefile [hunk 3]
```diff
@@ -1,10 +1,15 @@
-
 prefix	= /usr/local
+ifeq ($(OS),Windows_NT)
+	CC=gcc
+	LIBS=-lws2_32
```
Comment (djhaskin987) [inline Makefile:4]: On windows, we need to link against the winsock2 library.

**Comment by eraserhd** [comment]
Thanks!

**Pull request merged**